### PR TITLE
Kustomize optimization for Istio

### PR DESCRIPTION
**Pull request by gortegaMS**
# Type of PR

- [ ] Documentation changes
- [ ] Code changes
- [ ] Test changes
- [x] CI-CD changes
- [ ] GitHub Template changes

## PR Checklist

- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.
- [ ] My code follows the code style of this project.
- [x] I ran lint checks locally prior to submission.
- [x] Have you checked to ensure there aren't other open Pull Requests for the same update/change?

## Purpose of PR

Kustomize optimization for Istio

## Does this introduce a breaking change

- [ ] YES
- [x] NO

## Validation

- [ ] Unit tests updated and ran successfully
- [ ] Update documentation or issue referenced above

## Issues Closed or Referenced

- Closes https://github.com/retaildevcrews/wcnp/issues/1021

### CODE REVIEW

File: deploy/base/apps/istio/istio-ingress.yaml [hunk 1]
```diff
@@ -23,16 +23,16 @@ spec:
       k8s:
         resources:
           limits:
-            cpu: 200m
+            cpu: PLACEHOLDER
```
Comment (joaquinrz) [inline deploy/base/apps/istio/istio-ingress.yaml:26]: Can this PLACEHOLDER be something more descriptive? or was given this name for a reason?

File: deploy/dev-ngsa-asb-eastus/istio/kustomization.yaml [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+
+  - patch: |
+        - op: replace
+          path: "/spec/parameters/keyvaultName"
+          value: "kv-aks-3i2qzkkxofr7c"   
+    target:
+      kind: SecretProviderClass
+      name: istio-ingress-tls-secret-csi-akv
+      namespace: istio-system
+
```
Comment (joaquinrz) [inline deploy/dev-ngsa-asb-eastus/istio/kustomization.yaml:108]: line ending fix

File: deploy/dev-ngsa-asb-westus2/istio/kustomization.yaml [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+
+  - patch: |
+        - op: replace
+          path: "/spec/parameters/keyvaultName"
+          value: "kv-aks-3i2qzkkxofr7c"   
+    target:
+      kind: SecretProviderClass
+      name: istio-ingress-tls-secret-csi-akv
+      namespace: istio-system
+
```
Comment (joaquinrz) [inline deploy/dev-ngsa-asb-westus2/istio/kustomization.yaml:107]: line ending fix

File: deploy/pre-ngsa-asb-eastus/istio/kustomization.yaml [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+
+  - patch: |
+        - op: replace
+          path: "/spec/parameters/keyvaultName"
+          value: "kv-aks-fykg5bqasutle"   
+    target:
+      kind: SecretProviderClass
+      name: istio-ingress-tls-secret-csi-akv
+      namespace: istio-system
+
```
Comment (joaquinrz) [inline deploy/pre-ngsa-asb-eastus/istio/kustomization.yaml:70]: LE

File: deploy/pre-ngsa-asb-westus2/istio/kustomization.yaml [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+
+  - patch: |
+        - op: replace
+          path: "/spec/parameters/keyvaultName"
+          value: "kv-aks-fykg5bqasutle"   
+    target:
+      kind: SecretProviderClass
+      name: istio-ingress-tls-secret-csi-akv
+      namespace: istio-system
+
```
Comment (joaquinrz) [inline deploy/pre-ngsa-asb-westus2/istio/kustomization.yaml:70]: le

**Review by joaquinrz — APPROVED** [review]
Looks good, just some line ending fixes are needed and also had a questions regarding `PLACEHOLDER`

File: deploy/base/apps/istio/istio-ingress.yaml (hunk 1 above)
Comment (gortegaMS) [reply deploy/base/apps/istio/istio-ingress.yaml:26]: We agreed to use "PLACEHOLDER" on base files for now, any suggestions are welcome

File: deploy/dev-ngsa-asb-westus2/istio/kustomization.yaml (hunk 3 above)
Comment (gortegaMS) [reply deploy/dev-ngsa-asb-westus2/istio/kustomization.yaml]: Fixed

File: deploy/pre-ngsa-asb-eastus/istio/kustomization.yaml (hunk 4 above)
Comment (gortegaMS) [reply deploy/pre-ngsa-asb-eastus/istio/kustomization.yaml]: Fixed

File: deploy/pre-ngsa-asb-centralus/istio/kustomization.yaml [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+
+  - patch: |
+        - op: replace
+          path: "/spec/parameters/keyvaultName"
+          value: "kv-aks-fykg5bqasutle"   
+    target:
+      kind: SecretProviderClass
+      name: istio-ingress-tls-secret-csi-akv
+      namespace: istio-system
+
```
[the remark this answers is not in the document]
Comment (gortegaMS) [reply deploy/pre-ngsa-asb-centralus/istio/kustomization.yaml]: Fixed

File: deploy/pre-ngsa-asb-westus2/istio/kustomization.yaml (hunk 5 above)
Comment (gortegaMS) [reply deploy/pre-ngsa-asb-westus2/istio/kustomization.yaml]: Fixed

File: deploy/base/apps/istio/istio-ingress.yaml (hunk 1 above)
Comment (joaquinrz) [reply deploy/base/apps/istio/istio-ingress.yaml:26]: Cool, my question was more in regards of using the same PLACEHOLDER tag? But if it this was already reviewed and agreed with the team, I am good 👍🏻

**Pull request merged**